src/net/sourceforge/kolmafia/request/VolcanoIslandRequest.java

```diff
@@ -113,8 +114,15 @@ public static void getBreakfast() {
   }
 
   public static void parseResponse(final String urlString, final String responseText) {
-    if (!urlString.startsWith("volcanoisland.php") || !urlString.contains("action=tniat")) {
-      return;
+    if (!(urlString.startsWith("volcanoisland.php")
+        && (urlString.contains("action=tniat") || urlString.contains("action=npc")))) return;
```

Comment:
```suggestion
    if (!urlString.startsWith("volcanoisland.php")
        || !(urlString.contains("action=tniat") || urlString.contains("action=npc"))) return;
```

I think this is what you want -- only parse `volcanoisland.php`, and inside that only parse `tniat` and `npc` actions.

A test would ensure the logic is correct. Something like
```java
var cleanups = new Cleanups(withClass(AscensionClass.SAUCEROR), withProperty("_slimeVialsHarvested", 1));
try (cleanups)
 VolcanoIslandRequest.parseResponse("volcanoisland.php?action=npc", "You ladle some slime out of one of the drums. Fortunately, you had an empty vial on hand for just such an opportunity.");
 assertThat("_slimeVialsHarvested", isSetTo(2));
```
checking the URL is correct and perhaps including the full HTML.

Reply:
It's worked fine testing in game, but seems to be failing when I use that code for a test. Will have a look at it tomorrow, probably.

Reply:
I can add a test to your branch -- the first one can be tricky.